tenable/io/scans.py

```diff
@@ -173,11 +174,279 @@ def _create_scan_document(self, kw):
+        BYDAY or BYMONTHDAY in rrules. we will Split existing schedule details to
+        distributed schedule dictionary
+
+        Args:
+            details (dict): Dictionary of existing schedule
+
+        Returns:
+            :obj:`dict`:
+                Distributed dictionary of the keys required for scan schedule.
+        '''
+        schedule = {}
+        if details['rrules'] is not None:
+            rrules = dict(rules.split('=') for rules in details['rrules'].split(';'))
+            schedule = {
+                'frequency': rrules['FREQ'],
+                'interval': rrules['INTERVAL'],
+                'weekdays': rrules.get('BYDAY', '').split(),
+                'day_of_month': rrules.get('BYMONTHDAY', None),
+                'starttime': datetime.strptime(details['starttime'], '%Y%m%dT%H%M%S')
+                    if details['starttime'] is not None else None,
+                'timezone': details['timezone'] if details['timezone'] is not None else None,
+            }
+        return schedule
+
+    def create_scan_schedule(self, enabled=False, frequency=None, interval=None,
+                             weekdays=None, day_of_month=None, starttime=None, timezone=None):
+        '''
+        Create dictionary of keys required for scan schedule
+
+        Args:
+            enabled (bool, optional): To enable/disable scan schedule
+            frequency (str, optional):
+                The frequency of the rule. The string inputted will be up-cased.
+                Valid values are: ``ONETIME``, ``DAILY``, ``WEEKLY``,
+                ``MONTHLY``, ``YEARLY``.
+                Default value is ``ONETIME``.
+            interval (int, optional):
+                The interval of the rule.  The default interval is 1
+            weekdays (list, optional):
+                List of 2-character representations of the days of the week to
+                repeat the frequency rule on.  Valid values are:
+                *SU, MO, TU, WE, TH, FR, SA*
+                Default values: ``['SU', 'MO', 'TU', 'WE', 'TH', 'FR', 'SA']``
+            day_of_month (int, optional):
+                The day of the month to repeat a **MONTHLY** frequency rule on.
+                The default is today.
+            starttime (datetime, optional): When the scan should start.
+            timezone (str, optional):
+                The timezone to use for the scan.  The default if none is
+                specified is to use UTC.  For the list of usable timezones,
+                please refer to :devportal:`scans-timezones`
+
+        Returns:
+            :obj:`dict`:
+                Dictionary of the keys required for scan schedule.
+
+        '''
+        self._check('enabled', enabled, bool)
+        schedule = {}
+        if enabled is True:
+            launch = self._check('frequency', frequency, str,
+                 choices=['ONETIME', 'DAILY', 'WEEKLY', 'MONTHLY', 'YEARLY'],
```

Comment:
please have all the different "const" strings be declared once as a constant and shared across all the code instead of it being used multiple times as strings.
Examples:
"ONETIME" - shows 15 times in the code.
"BYDAY" - shows 7 times in the code.
"FREQ" - shows 4 times in the code. 
etc.

Reply:
Added constants file and replaced repeating values with const variables

Reply:
If the concern is DRY, then why not just pull out the input validation into a constructor like in the SC package and centralize all of this logic?